### PR TITLE
Manager: Fix configuring UI features through URL parameters

### LINKED ISSUE #23550: [Bug]: Unable to configure features through URL parameters
### Describe the bug

Unable to configure features (full screen, nav, panel, etc ...) through URL parameters: https://storybook.js.org/docs/react/configure/features-and-behavior#configuring-through-url-parameters

### To Reproduce

Trying to set panel position through URL parameters (I am using Yann's repo as example):

V7.1 (Not working):

- Right Panel: https://mealdrop.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=right
- Bottom Panel: https://mealdrop.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=bottom
- No Panel: https://mealdrop.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=false

V6.4.7 (Working):

- Right Panel: https://mealdrop-git-feat-fix-test-ci-yannbf.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=right
- Bottom Panel: https://mealdrop-git-feat-fix-test-ci-yannbf.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=bottom
- No Panel: https://mealdrop-git-feat-fix-test-ci-yannbf.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=false

_(Try to open link in Incognito mode for "best" result)_

---

The same issue applies to [other features/query params](https://storybook.js.org/docs/react/configure/features-and-behavior#configuring-through-url-parameters) (`shortcuts`, `full`, `nav`, `panel`, etc ...)

### System

```shell
Environment Info:

  System:
    OS: macOS 13.4.1
    CPU: (12) arm64 Apple M2 Max
  Binaries:
    Node: 16.19.1 - ~/.nvm/versions/node/v16.19.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.19.1/bin/yarn
    npm: 8.19.3 - ~/.nvm/versions/node/v16.19.1/bin/npm
  Browsers:
    Safari: 16.5.2
  npmPackages:
    @storybook/addon-a11y: ^7.1.0 => 6.5.13 
    @storybook/addon-actions: ^7.1.0 => 6.5.13 
    @storybook/addon-essentials: ^7.1.0 => 6.5.13 
    @storybook/addon-interactions: ^7.1.0 => 6.5.13 
    @storybook/addon-links: ^7.1.0 => 6.5.13 
    @storybook/mdx2-csf: ^1.1.0 => 0.0.3 
    @storybook/preset-scss: ^1.0.3 => 1.0.3 
    @storybook/testing-library: 0.2.0 => 0.0.13 
    @storybook/vue3: ^7.1.0 => 6.5.13
```

### Additional context

We are embedding Storybook in other website, and with lacking options to hide side panel through URL parameters is causing us to show unrelated/irrelevant information on the website. And not able to change the panel position is also causing layout issue to us.

### PR DESCRIPTION
**Pull request by limhenry**
Closes #23550

## What I did

Fix unable to configure features (full screen, nav, panel, etc ...) through URL parameters

## How to test

1. Run `yarn start`
2. Open Storybook in your browser
3. Should able to change panel position from bottom to right using URL Parameters (`&panel=right`, `&panel=bottom`, ...)

## Example links 
- http://localhost:6006/?path=/story/example-button--primary&panel=bottom
- http://localhost:6006/?path=/story/example-button--primary&panel=right
- http://localhost:6006/?path=/story/example-button--primary&panel=false

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by kasperpeulen — COMMENTED** [review]
Can you explain why this was needed to me?

**Comment by limhenry** [comment]
@kasperpeulen 

> Can you explain why this was needed to me?

TL:DR: `api.setOptions(merge(api.getInitialOptions(), persisted))` is overriding the value from url.ts

Full explanation: https://github.com/storybookjs/storybook/issues/23550#issuecomment-1661064451

**Review by kasperpeulen — APPROVED** [review]
Okay, seems legit!

File: code/lib/manager-api/src/modules/layout.ts [hunk 1]
```diff
@@ -283,7 +283,6 @@ export const init: ModuleFn = ({ store, provider, singleStory, fullAPI }) => {
     api,
     state: merge(api.getInitialOptions(), persisted),
     init: () => {
-      api.setOptions(merge(api.getInitialOptions(), persisted));
```
Comment (kylegach) [inline code/lib/manager-api/src/modules/layout.ts:286]: Hey @ndelangen, can you weigh in on this change, please? The removed line was [introduced](https://github.com/storybookjs/storybook/pull/18550/files#diff-a8e2c0efe97f49b3d7be7fcd346a40715f169f63bdb513d0e45e51af07f3c221R303) in your PR, per [this comment](https://github.com/storybookjs/storybook/issues/23550#issuecomment-1661064451).

Our understanding is that these options were already set at this point by `url.ts`, and this line clobbers those.

Comment (kasperpeulen) [reply code/lib/manager-api/src/modules/layout.ts:286]: @limhenry FYI, the one with most knowledge about this line is on holiday, we will come back to to this next week!

**Review by JReinhold — APPROVED** [review]
Tried this locally, and this seems correct to me.

**Review by ndelangen — CHANGES_REQUESTED** [review]

**Comment by limhenry** [comment]
```
api.setOptions(merge(merge(api.getInitialOptions(), persisted), LayoutOptionsFromURL));
```

What's the best way for me to get the value for `LayoutOptionsFromURL` (from url.ts)?

@ndelangen @yannbf

**Comment by ndelangen** [comment]
`layout.ts` knows what query parameters to look for, my suggestion would be implement this in `layout.ts` itself.

**Comment by kimtuck** [comment]
This feature works in 6.x, but is broken in 7.x.  This is really useful for visual regression testing -- comparing a screenshot of the story against a previous screenshot.  If the new screenshot contains a new story, that story is shown in the sidebar, which causes the screenshots to be different and fails the test.

By adding these parameters (&full=true&nav=false&panel=false&tabs=false) the screenshot contains only the story's contents; and is unaffected by adding or removing stories.

If no one is working on this issue, I can take a look.  Thanks.

**Comment by limhenry** [comment]
> This feature works in 6.x, but is broken in 7.x. This is really useful for visual regression testing -- comparing a screenshot of the story against a previous screenshot. If the new screenshot contains a new story, that story is shown in the sidebar, which causes the screenshots to be different and fails the test.
> 
> By adding these parameters (&full=true&nav=false&panel=false&tabs=false) the screenshot contains only the story's contents; and is unaffected by adding or removing stories.
> 
> If no one is working on this issue, I can take a look. Thanks.

It looks like it is fixed in latest dev branch (8.0.0-alpha.15) 😄

**Comment by ndelangen** [comment]
Looks that way @limhenry ! 😮 

Sorry, but this PR has no changes left, so I'll close it.

We appreciate the time, energy and effort you put into this PR.

**Pull request closed without merging**